### PR TITLE
DOCS: Extending Ice interface example

### PR DESCRIPTION
**Pull request by KetchupBomb**
In order to help document & maintain tribal knowledge, include a few snippets regarding testing Ice interface changs.

### Checks

- [x] My commits follow the [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

### CODE REVIEW

**Comment by KetchupBomb** [comment]
@Krzmbrzl, view [rendered doc on my branch](https://github.com/KetchupBomb/mumble/blob/docs-ice-interface-example/docs/dev/ExtendingTheIceInterface.md#testing-ice-interface-changes). Is it too wordy? This is the level of detail I'd want for myself, but let me know if you want it trimmed down.

**Review by Krzmbrzl — APPROVED** [review]
LGTM :+1:

**Comment by Krzmbrzl** [comment]
Thanks for writing this :+1:
